code/modules/detectivework/scanner.dm

```diff
@@ -30,16 +60,73 @@
 		scanner.displayDetectiveScanResults(usr)
 
 /obj/item/detective_scanner/attack_self(mob/user)
+	if(scanning)
+		to_chat(user, "<span class='notice'>[src] is in use.</span>")
+		return
+	radial_generate()
+	if(icons_available)
+		var/selection = show_radial_menu(user, src, icons_available, radius = 38, require_near = TRUE, tooltips = TRUE)
+		if(!selection)
+			return
+		
+		if(selection == "View results")
+			displayDetectiveScanResults(user)
+			return
+		
+		if(selection == "Volume on" || selection == "Volume off")
+			to_chat(user, (sound_on ? "<span class='notice'>You mute the volume.</span>" : "<span class='notice'>You turn up the volume.</span>"))
+			sound_on = !sound_on
+			return
+
+		if(selection == "Print results")
+			option_print(user)
+			return
+
+		if(selection == "Clear results")
+			option_clearlogs(user)
+			return
+	
+/obj/item/detective_scanner/proc/option_print(mob/user)
 	if(log.len && !scanning)
 		scanning = 1
+		scan_animation()
 		to_chat(user, "<span class='notice'>Printing report, please wait...</span>")
-		addtimer(CALLBACK(src, .proc/PrintReport), 100)
+		if(admin)
+			PrintReport() // admin scanner bypasses print wait
+		else
+			addtimer(CALLBACK(src, .proc/PrintReport), (scan_speed * 25) )
 	else
 		to_chat(user, "<span class='notice'>The scanner has no logs or is in use.</span>")
 
+/obj/item/detective_scanner/proc/option_clearlogs(mob/user)
+	if(!user.canUseTopic(src, be_close=TRUE) && !admin) // Best way for checking if a player can use while not incapacitated, etc. Admin scanner doesn't care
+		return
+	if(!LAZYLEN(log))
+		to_chat(user, "<span class='notice'>Cannot clear logs, the scanner has no logs.</span>")
+		return
+	if(scanning)
+		to_chat(user, "<span class='notice'>Cannot clear logs, the scanner is in use.</span>")
+		return
+	to_chat(user, "<span class='notice'>The scanner logs are cleared.</span>")
+	log.Cut()
+
 /obj/item/detective_scanner/attack(mob/living/M, mob/user)
 	return
 
+/obj/item/detective_scanner/proc/scan_animation()
+	if(scan_icon)
+		icon_state = (scanning ? icon_state_scanning : icon_state_neutral)
+
+/obj/item/detective_scanner/proc/radial_generate() // generate option menu in order to show context sensitive menu
+	icons_available = list()
+	icons_available += list("View results" = image(icon = icon_directory, icon_state = "view"))
+	if(can_sound)
+		sound_on ? (icons_available += list("Volume off" = image(icon = icon_directory, icon_state = "volume_off"))) : (icons_available += list("Volume on" = image(icon = icon_directory, icon_state = "volume_on")))
```

Comment:
these can probably be smooshed down into one option called toggle sound which changes its image based on whether sound is on, but it's not really that important